src/compiler/program.ts

```diff
@@ -724,6 +741,10 @@ namespace ts {
                         }
                         break;
                 }
+
+                if (isSourceFileJavaScript(file)) {
+                    forEachChild(node, node => collect(node, allowRelativeModuleNames));

```

Comment:
This isn't quite right, you'll end up looking for ES6 module constructs in nested locations where they aren't allowed. You really only should be looking for call expressions in the child nodes.